homeassistant/components/group/light.py

```diff
@@ -59,6 +60,7 @@
     {
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_ICON, default="mdi:lightbulb-group"): cv.icon,
```

Comment:
Isn't it possible to set an icon for the entity in lovelace frontend?

Reply:
It is, but the root [group integration ](https://www.home-assistant.io/integrations/group/) allows for setting an icon in YAML, yet none of the other groups do. Imo it's a bit tidier to allow setting it in YAML than using customize.yaml

Reply:
We don't want frontend options in the backend. The general group has a lot of legacy. The user can set a unique id for the light group and be able to change icon in the frontend or use the lovelace frontend for full customization.

Reply:
This seems inconsistent with many other integrations (Slack notify, webostv, generic group, fibaro, netdata), do they need updating for the icon parameter to be removed then?

Reply:
All of those are legacy features. We don't need to remove them, just don't add any more of them.